conda/lock.py

```diff
@@ -61,11 +73,44 @@ def __enter__(self):
                 return self
 
         stdoutlog.error("Exceeded max retries, giving up")
-        raise LockError(lockstr % self.lock_path)
+        raise LockError(LOCKSTR % self.lock_path)
 
     def __exit__(self, exc_type, exc_value, traceback):
         try:
             os.rmdir(self.lock_path)
             os.rmdir(self.path)
         except OSError:
             pass
+
+
+class FileLock(object):
+    """
+    Context manager to handle locks.
+    """
```

Comment:
I didn't update anything with the docstring, and the current one sucks.  That's bad.

All new code that is meant to be "public" outside the module should be documented with google-style docstrings.
http://www.sphinx-doc.org/en/stable/ext/example_google.html

Also any "private" code you feel like it's important to document for clarity.